src/Microsoft.AspNetCore.BuildTools.Tasks/GetGitCommitInfo.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    public class GetGitCommitInfo : Task
+    {
+        /// <summary>
+        /// A folder inside the git project. Does not need to be the top folder.
+        /// This task will search upwards for the .git folder.
+        /// </summary>
+        [Required]
+        public string WorkingDirectory { get; set; }
+
+        /// <summary>
+        /// The name of the branch HEAD is pointed to. Can be null or empty
+        /// for repositories in detached HEAD mode.
+        /// </summary>
+        [Output]
+        public string Branch { get; set; }
+
+        /// <summary>
+        /// The full commit SHA of the current commit referenced by HEAD.
+        /// </summary>
+        [Output]
+        public string CommitHash { get; set; }
+
+        /// <summary>
+        /// The folder containing the '.git', not the .git folder itself.
+        /// </summary>
+        [Output]
+        public string GitProjectRootPath { get; set; }
+
+        public override bool Execute()
+        {
+            Branch = null;
```

Comment:
MSBuild shouldn't instantiate these properties since they're marked as 'output'. So yeah, can be removed.